contrib/gimp.lua

```diff
@@ -0,0 +1,179 @@
+    and opens the files.  After editing, the exported images are overwritten to save the 
+    changes.  When gimp exits, the exported files are moved into the current collection
+    and imported into the database.  The imported files then show up grouped with the 
+    originally selected images.
+
+    ADDITIONAL SOFTWARE NEEDED FOR THIS SCRIPT
+    * gimp - http://www.gimp.org
+
+    USAGE
+    * require this script from your main lua file
+    * select an image or images for editing with gimp
+    * in the export dialog select "Edit with gimp" and select the format and bit depth for the
+      exported image
+    * Press "export"
+    * Edit the image with gimp then save the changes with File->Overwrite....
+    * Exit gimp
+    * The edited image will be imported and grouped with the original image
+
+    CAVEATS
+    * Developed and tested on Ubuntu 14.04 LTS with darktable 2.0.3 and gimp 2.9.3 (development version with
+      > 8 bit color)
+    * There is no provision for dealing with the xcf files generated by gimp, since darktable doesn't deal with 
+      them.  You may want to save the xcf file if you intend on doing further edits to the image or need to save 
+      the layers used.  Where you save them is up to you.
+
+    BUGS, COMMENTS, SUGGESTIONS
+    * Send to Bill Ferguson, wpferguson@gmail.com
+]]
+
+local dt = require "darktable"
+local gettext = dt.gettext
+
+dt.configuration.check_version(...,{3,0,0})
+
+-- Tell gettext where to find the .mo file translating messages for a particular domain
+gettext.bindtextdomain("gimp",dt.configuration.config_dir.."/lua/")
+
+-- Thanks to http://lua-users.org/wiki/SplitJoin for the split and split_path functions
+local function split(str, pat)
+   local t = {}  -- NOTE: use {n = 0} in Lua-5.0
+   local fpat = "(.-)" .. pat
+   local last_end = 1
+   local s, e, cap = str:find(fpat, 1)
+   while s do
+      if s ~= 1 or cap ~= "" then
+        table.insert(t,cap)
+      end
+      last_end = e+1
+      s, e, cap = str:find(fpat, last_end)
+   end
+   if last_end <= #str then
+      cap = str:sub(last_end)
+      table.insert(t, cap)
+   end
+   return t
+end
+
+local function split_path(str)
+   return split(str,'[\\/]+')
+end
+
+local function get_filename(str)
+  parts = split_path(str)
+  return parts[#parts]
+end
+
+local function basename(str)
+  return string.sub(str,1,-4)
+end
+
+local function _(msgid)
+    return gettext.dgettext("gimp", msgid)
+end
+
+local function checkIfBinExists(bin)
+  local handle = io.popen("which "..bin)
+  local result = handle:read()
+  local ret
+  handle:close()
+  if (result) then
+    dt.print_error("true checkIfBinExists: "..bin)
+    ret = true
+  else
+    dt.print_error(bin.." not found")
+    ret = false
+  end
+  return ret
+end
+
+local function show_status(storage, image, format, filename,
+  number, total, high_quality, extra_data)
+  dt.print("Exporting to gimp "..tostring(number).."/"..tostring(total))
```

Comment:
I had a fist look and one thing that could be improved is to mark this string for translation.
Have a look at "kml_export.lua" line 72:
https://github.com/darktable-org/lua-scripts/blob/master/contrib/kml_export.lua#L72 
> dt.print(string.format(_("Export Image %i/%i"), number, total))